client/rest/build.gradle

```diff
@@ -38,13 +55,50 @@ publishing {
   }
 }
 
+configurations {
+  shade {
+    transitive = false
+  }
+}
+
+// Useful for build time dependencies, as it is generated before compilation of the source in the rest client.
+// This cannot be used as the final shaded jar, as it will contain the compiled source and dependencies
+File shadedDir = file("${buildDir}/shaded")
+File shadedSrcDir = file("${buildDir}/generated-dummy-shaded")
+task shadeDeps(type: com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar) {
+  destinationDir = shadedDir
+  configurations = [project.configurations.shade]
+  classifier = 'deps'
+  relocate 'org.apache', 'internal.org.apache'
+
+  doLast {
+    shadedSrcDir.mkdir()
+  }
+}
+
+// builds the actual shaded jar that is being distributed
+shadowJar {
+  configurations = [project.configurations.shade]
+  classifier = null
+  relocate 'org.apache', 'internal.org.apache'
```

Comment:
Shouldn't there be no need to use `relocate` here? If there is then unit tests ran with unshaded classes, which is exactly what we were trying to avoid.